tests/foreman/cli/test_host.py

```diff
@@ -3073,8 +3074,13 @@ def _register_client(self, activation_key=None, lce=False,
         :param auto_attach: boolean to indicate whether to register with
             auto-attach option, in case of registration with activation key a
             command is launched
+        :param attach_to_default: boolean to indicate whether to attach to
+            plain RHEL subsctiption
         :return: the registration result
         """
+        assert not auto_attach or not attach_to_default, \
+            'Only one of auto_attach or attach_to_default can be set'
```

Comment:
1. 
```suggestion
            'Only one of auto_attach or attach_to_default must be set'
```
2. Due to above validation `self._register_client` will not work for other tests where it used i.e. https://github.com/SatelliteQE/robottelo/blob/cc4a6efe766fbe68eef99646c300287d46b7dcc6/tests/foreman/cli/test_host.py#L3208
please check
3. Why do we require it ?

Reply:
If none of these is set, assert should pass:
```
>>> auto_attach=True; attach_to_default=True; not auto_attach or not attach_to_default
False
>>> auto_attach=False; attach_to_default=True; not auto_attach or not attach_to_default
True
>>> auto_attach=True; attach_to_default=False; not auto_attach or not attach_to_default
True
>>> auto_attach=False; attach_to_default=False; not auto_attach or not attach_to_default
True
```

Reply:
Why is this needed: Test auto attaches and then attempts to enable `rhel-7-server-satellite-capsule-6.2-rpms` and asserts that repo is not enabled (because normal RHEL subscription does not provide capsule). Problem was auto attach was giving me subscription that had that entitlement, so repo enable passed. This way I explicitly request RHEL subscription, making test result more stable.